checkov/azure_pipelines/common/resource_id_utils.py

```diff
@@ -0,0 +1,40 @@
+from __future__ import annotations
+
+from typing import Any, Dict, List
+
+from checkov.common.util.consts import START_LINE, END_LINE
+
+
+def generate_resource_key_recursive(start_line: int, end_line: int,
+                                    file_conf: Dict[str, Any] | List[Dict[str, Any]], resource_key: str | None = None
+                                    ) -> str | None:
+    if not isinstance(file_conf, dict):
+        return resource_key
+
+    def _get_resource_from_code_block(block_to_inspect: dict[str, Any], inspected_key: str | None) -> str | None:
+        if block_to_inspect[START_LINE] <= start_line <= end_line <= block_to_inspect[END_LINE]:
+            block_name = block_to_inspect.get('displayName',
+                                              block_to_inspect.get('name',
+                                                                   block_to_inspect.get('job',
+                                                                                        block_to_inspect.get('stage',
+                                                                                                             False))))
+            inspected_key = f'{inspected_key}({block_name})' if block_name else inspected_key
+            if block_to_inspect[START_LINE] == start_line:
+                return inspected_key
+            return generate_resource_key_recursive(start_line, end_line, block_to_inspect, resource_key=inspected_key)
+        return None
```

Comment:
it would be better to move the function to the root file level instead of using an inline function. you will need to pass `start_line` and `end_line` as normal as normal parameter. Currently the function is recreated each time you call `generate_resource_key_recursive()`